packages/rpc-core/src/rpc-methods/__tests__/get-fee-for-message-test.ts

```diff
@@ -0,0 +1,144 @@
+import type { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import { Blockhash } from '@solana/transactions';
+import fetchMock from 'jest-fetch-mock-fork';
+
+import { Commitment } from '../common';
+import { createSolanaRpcApi, SolanaRpcMethods } from '../index';
+
+// See scripts/fixtures/send-transaction-fee-payer.json
+const MOCK_PUBLIC_KEY_BYTES = // DRtXHDgC312wpNdNCSb8vCoXDcofCJcPHdAw4VkJ8L9i
+    // prettier-ignore
+    new Uint8Array([
+        0xb8, 0xac, 0x70, 0x4f, 0xaf, 0xc7, 0xa5, 0xfc, 0x8c, 0x5d, 0x1f, 0x0a, 0xc8, 0xcf, 0xaa, 0xe0,
+        0x42, 0xfa, 0x3b, 0xb8, 0x25, 0xf0, 0xec, 0xfc, 0xe2, 0x27, 0x4d, 0x7d, 0xad, 0xad, 0x51, 0x2d,
+    ]);
+
+function getMockTransactionMessage(blockhash: Blockhash) {
+    const memoString = 'Hello from the web3.js tests!';
+    const blockhashBytes = fixSerializer(base58, 32).serialize(blockhash);
+    // prettier-ignore
+    const message = new Uint8Array([
+        /** VERSION HEADER */
+        0x80, // 0 + version mask
+
+        /** MESSAGE HEADER */
+        0x01, // numSignerAccounts
+        0x00, // numReadonlySignerAccount
+        0x01, // numReadonlyNonSignerAccounts
+
+        /** STATIC ADDRESSES */
+        0x02, // Number of static accounts
+            ...MOCK_PUBLIC_KEY_BYTES,
+            0x05, 0x4a, 0x53, 0x5a, 0x99, 0x29, 0x21, 0x06, 0x4d, 0x24, 0xe8, 0x71, 0x60, 0xda, 0x38, 0x7c, 0x7c, 0x35, 0xb5, 0xdd, 0xbc, 0x92, 0xbb, 0x81, 0xe4, 0x1f, 0xa8, 0x40, 0x41, 0x05, 0x44, 0x8d, // MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr
+
+        /** TRANSACTION LIFETIME TOKEN (ie. the blockhash) */
+        ...blockhashBytes,
+
+        /* INSTRUCTIONS */
+        0x01, // Number of instructions
+
+            // First instruction
+            0x01, // Program address index
+            0x00, // Number of address indices
+            memoString.length, // Length of instruction data
+                ...new TextEncoder().encode(memoString),
+
+        /** ADDRESS TABLE LOOKUPS */
+        0x00, // Number of address table lookups
+    ]);
+    const messageBase64 = base64.deserialize(message)[0];
+    return messageBase64;
+}
+
+describe('getFeeForMessage', () => {
+    let rpc: Rpc<SolanaRpcMethods>;
+    beforeEach(() => {
+        fetchMock.resetMocks();
+        fetchMock.dontMock();
+        rpc = createJsonRpc<SolanaRpcMethods>({
+            api: createSolanaRpcApi(),
+            transport: createHttpTransport({ url: 'http://127.0.0.1:8899' }),
+        });
+    });
+
+    (['confirmed', 'finalized', 'processed'] as Commitment[]).forEach(commitment => {
+        describe(`when called with \`${commitment}\` commitment`, () => {
+            if (commitment === 'finalized') {
+                it.todo(
+                    'returns the result as a bigint (test broken; see https://discord.com/channels/428295358100013066/560496939779620864/1132048104728825926)'
+                );
+                return;
+            }
```

Comment:
Hmm is this a timing issue that could affect `confirmed` too if the test runs early enough?

Reply:
I'm not sure on confirmed. I plucked this from https://github.com/solana-labs/solana-web3.js/blob/master/packages/rpc-core/src/rpc-methods/__tests__/send-transaction-test.ts#L115-L120

It's definitely related to the fact that it takes ~17s for the test validator to finalize slot 0, but `confirmed` seems to be almost immediate so maybe that's not a concern? I don't see `confirmed` omitted anywhere else in our test suites